src/VisualStudio/Core/Def/ServicesVSResources.resx

```diff
@@ -2031,4 +2031,7 @@ Additional information: {1}</value>
   <data name="Sync_namespaces_changes" xml:space="preserve">
     <value>Sync namespaces changes</value>
   </data>
+  <data name="Enable_document_outline_experimental_requires_restart" xml:space="preserve">
+    <value>Enable document outline (experimental, requires restart)</value>
```

Comment:
Given you're only checking the flag on GetOutline, I suspect you'd have to reopen the file only. Might be worth changing the text so that way the user doesn't think they have to restart all of VS.

Reply:
tbh, i'm really fine with all experimental flags telling users to reopen/close vs.  it just is the simplest model and means less complex logic/invariants for any experimental feature :)